### PR TITLE
fix(ext/node): support AES-128-ECB cipher for PKCS#8 private key encryption and decryption

### PR DESCRIPTION
**Pull request by divybot**
## Summary

- The `pkcs5` crate does not implement AES-128-ECB in its PBES2 support, so `generateKeyPairSync` with `privateKeyEncoding: { cipher: 'aes-128-ecb' }` was throwing `TypeError: Unsupported cipher for PKCS#8 encryption: aes-128-ecb`
- Adds manual DER construction for `EncryptedPrivateKeyInfo` using PBES2+PBKDF2-HMAC-SHA256+AES-128-ECB, plus a matching manual DER parser for decryption
- Enables `parallel/test-crypto-publicDecrypt-fails-first-time.js` node compat test

## Test plan

- [x] `cargo test --test node_compat -- "publicDecrypt" --nocapture` passes
- [x] Pre-existing `test-crypto-key` failures are identical before and after this change (verified by stash/unstash)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by fibibot — COMMENTED** [review]
Substance walks correct on a careful read of the custom DER builder/parser + PBES2 round-trip. CI failure is the recurring Windows DNS flake (`test-dns-resolver-max-timeout.js`), unrelated to this PR.

## What's right

- **DER builder primitives** (`der_tlv`, `der_seq`, `der_oct`, `der_oid_tlv`, `der_null`, `der_uint`): standard ASN.1 patterns. Length encoding handles short-form, 0x81 single-byte long-form, and 0x82 two-byte long-form — sufficient for EncryptedPrivateKeyInfo blobs which won't exceed 64KB. `der_uint` correctly prepends a zero byte for high-bit-set values (DER's sign-bit convention).
- **`parse_der_tlv`** has solid bounds checks: `data.len() < 2`, `data.len() < 3` for 0x81, `data.len() < 4` for 0x82, and the final `data.len() < end` after computing length. Each chained call returns `None` on malformed input, which short-circuits via `?`. No panic risk on adversarial input.
- **`build_pbes2_aes128ecb_epki`**: EncryptedPrivateKeyInfo structure correctly nests PBES2 OID → PBKDF2 alg (with SHA256 + iterations + salt + PRF) → AES-128-ECB alg (OID only, no params since ECB has no IV).
- **`try_decrypt_pbes2_aes128ecb`**: validates each OID at the right nesting level (PBES2, PBKDF2, HMAC-SHA256 implied via PRF, AES-128-ECB). Returns `None` (not error) on non-AES-128-ECB blobs so the caller falls through to `pkcs5`'s general decoder for other ciphers.
- **2048 PBKDF2 iterations** matches OpenSSL's default — same number used elsewhere in the codebase for PBES2 export.
- **`pem_to_der_if_label`** correctly validates the PEM label before base64-decoding the body.

## Concerns worth resolving

### 1. AES-128-ECB is a weak mode — needs a SECURITY comment

ECB mode is widely discouraged because identical plaintext blocks produce identical ciphertext (no randomization). For private-key data this is mostly fine (keys are themselves random), but the *general* recommendation against ECB is well-known, and a future reader looking at the code should understand this is supported for **Node-compat with legacy keys** rather than recommended for new keys.

Worth a `// SECURITY:` comment block on `encrypt_pkcs8_aes128ecb` explaining that AES-128-ECB is supported for compatibility with Node-generated keys (and to enable round-tripping legacy artifacts), and that new keys should use AES-128-CBC or stronger. The Vaudenay padding-oracle concern that bites CBC mode doesn't apply to ECB (no chained-block structure), so unlike #33762/#33769 the constant-time-unpad fix isn't relevant here — but the format-level weaknesses are still worth documenting.

### 2. No Rust-level round-trip unit test

The Node compat test (`parallel/test-crypto-publicDecrypt-fails-first-time.js`) exercises an end-to-end JS scenario, but a Rust unit test that:

```rust
#[test]
fn test_pbes2_aes128ecb_roundtrip() {
  let plaintext = b"some pkcs8 der bytes";
  let encrypted = encrypt_pkcs8_aes128ecb(plaintext, b"password");
  let decrypted = try_decrypt_pbes2_aes128ecb(&encrypted, b"password").unwrap();
  assert_eq!(plaintext, decrypted.as_slice());
}
```

would lock in the encode/decode parity at the DER layer. The current test would catch a regression in either side, but a unit test isolates which side broke.

Also worth a negative test: `try_decrypt_pbes2_aes128ecb` with the wrong passphrase returns `None` (the `decrypt_padded_vec_mut::<Pkcs7>` should error on bad PKCS#7 padding from a wrong-key decrypt, propagating to `.ok()` returning `None`).

### 3. Custom DER + crypto = audit surface

286 lines of hand-rolled ASN.1 DER plus PBES2 plumbing. The walk above checks out, but the same general concern as #33089's custom AES-CCM applies: custom crypto/ASN.1 code is the kind of thing where edge cases bite later. If the `pkcs5` crate ever adds AES-ECB support, dropping this code in favor of it would eliminate the audit surface.

Not blocking — `pkcs5` may never add ECB support given how discouraged it is, and the implementation is correct enough that this is a genuine choice between "custom code now" vs "wait for upstream".

## CI

86 pass, 18 pending, 1 fail. The lone failure is `test node_compat (3/3) debug windows-x86_64` on `parallel/test-dns-resolver-max-timeout.js` — recurring Windows DNS-timing flake, unrelated to crypto changes.

APPROVE-shape from my side modulo the SECURITY comment and ideally the round-trip unit test.

File: ext/node_crypto/keys.rs [hunk 1]
```diff
@@ -3376,6 +3392,246 @@ pub enum ExportPrivateKeyPemError {
...
+const OID_PBES2_BYTES: &[u8] =
+  &[0x2a, 0x86, 0x48, 0x86, 0xf7, 0x0d, 0x01, 0x05, 0x0d]; // 1.2.840.113549.1.5.13
+const OID_PBKDF2_BYTES: &[u8] =
+  &[0x2a, 0x86, 0x48, 0x86, 0xf7, 0x0d, 0x01, 0x05, 0x0c]; // 1.2.840.113549.1.5.12
+const OID_HMAC_SHA256_BYTES: &[u8] =
+  &[0x2a, 0x86, 0x48, 0x86, 0xf7, 0x0d, 0x02, 0x09]; // 1.2.840.113549.2.9
+const OID_AES128_ECB_BYTES: &[u8] =
+  &[0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x01, 0x01]; // 2.16.840.1.101.3.4.1.1
+
+fn der_tlv(tag: u8, value: &[u8]) -> Vec<u8> {
```
Comment (littledivy) [inline ext/node_crypto/keys.rs]: Can we use the asn1 and der crates instead?

Comment (divybot) [reply ext/node_crypto/keys.rs]: Done — the `spki::der` crates (`AnyRef`, `OctetStringRef`, `Encode`, `AlgorithmIdentifierRef`, `Pbkdf2Params`, etc.) are already used throughout the implementation. The earlier draft used hand-rolled `der_tlv`/`der_seq` helpers; those were replaced with the `spki::der` API in a previous commit.
